docs/oathkeeper/guides/websockets.mdx

```diff
@@ -0,0 +1,360 @@
+  upstream:
+    preserve_host: true
+    url: "http://ws:8080"
+  match:
+    url: "http://127.0.0.1:8080/<**>"
+    methods:
+      - GET
+      - POST
+  authenticators:
+    - handler: cookie_session
+  mutators:
+    - handler: noop
+  authorizer:
+    handler: allow
+  errors:
+    - handler: redirect
+      config:
+        to: http://127.0.0.1:4455/login
+```
+
+This configuration of Oathkeeper uses cookie authenticator against Ory Kratos and proxies only authenticated requests to `http://ws:8080` upstream. The `ws` hostname is resolved through the docker network. If you aren't deploying your application within docker, this would just be your localhost IP.
+
+## WebSocket service
+
+1. Let's create a folder `ws` and create our websocket service using Go and Gin framework. Create `ws/main.go` file with the following content:
+
+```go
+package main
+
+import (
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/websocket"
+)
+
+func main() {
+
+	r := gin.Default()
+	r.LoadHTMLFiles("index.html")
+
+	r.GET("/", func(c *gin.Context) {
+		c.HTML(200, "index.html", nil)
+		return
+	})
+
+	r.GET("/ws", func(c *gin.Context) {
+		var wsupgrader = websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+		}
+		conn, err := wsupgrader.Upgrade(c.Writer, c.Request, nil)
+		if err != nil {
+			fmt.Printf("Failed to set websocket upgrade: %+v\n", err)
+			return
+		}
+
+		for {
+			t, msg, err := conn.ReadMessage()
+			if err != nil {
+				break
+			}
+			conn.WriteMessage(t, msg)
+		}
+		return
+	})
+
+	r.Run(":8080")
+}
+
+```
+2. We need to initialize go modules by running the following commands:
+
+```shell
+cd ws
+go mod init ws
+go mod tidy
+```
+
+3. Create `ws/index.html` file with the following content:
+
+```html
+<html>
+<head>
+    <script src="https://code.jquery.com/jquery-2.1.1.min.js"></script>
+</head>
+
+<body>
+<h3>WebSocket Go</h3>
+<pre id="output"></pre>
+
+<script>
+    url = 'ws://127.0.0.1:8080/ws';
+    c = new WebSocket(url);
+
+    send = function(data){
+        $("#output").append((new Date())+ " ==> "+data+"\n")
+        c.send(data)
+    }
+
+    c.onmessage = function(msg){
+        $("#output").append((new Date())+ " <== "+msg.data+"\n")
+        console.log(msg)
+    }
+
+    c.onopen = function(){
+        setInterval(
+            function(){ send("ping") }
+            , 1000 )
+    }
+</script>
+
+</body>
+</html>
+```
+
+4. Create `ws/Dockerfile` with the following content
+
+```
+FROM golang as builder
+
+RUN mkdir /build
+
+ADD . /build
+
+WORKDIR /build
+RUN GOOS=linux GOARCH=amd64 go build -o ws main.go
+
+FROM alpine
+EXPOSE 8090
+
+COPY --from=builder /build/ws /ws
+COPY index.html /index.html
+ENTRYPOINT ["/ws"]
+```
+5. We need to add our `ws` service to the `docker-compose.yml`
+
+```yaml
+services:
+...
+  ws:
+    build:
+      context: "ws"
+    networks:
+      - intranet
+...
+```
+
+## Testing
+
+We can simply run `docker-compose up` wait for services to be ready and
```

Comment:
There are 3 actions in this sentence and no punctuation at all - hurts my eyes :D